### PR TITLE
CONSOLE 3070: Console-operator should pass infrastructure config's ControlPlaneToplogy to the console-config.yaml

### PR DESCRIPTION
**Pull request by jhadvig**
Infrastructure config's `ControlPlaneToplogy` will be stored in the console-config.yaml under `clusterInfo.controlPlaneTopology`.

/assign @zherman0

### CODE REVIEW

**Review by zherman0 — APPROVED** [review]

**Comment by jhadvig** [comment]
/retest

```
[36mINFO[0m[2022-02-23T12:11:38Z] Step e2e-aws-operator-gather-aws-console succeeded after 1m10s. 
[36mINFO[0m[2022-02-23T12:11:38Z] Running step e2e-aws-operator-gather-must-gather. 
[36mINFO[0m[2022-02-23T12:25:08Z] Logs for container test in pod e2e-aws-operator-gather-must-gather: 
[36mINFO[0m[2022-02-23T12:25:08Z] Running must-gather...


Error running must-gather collection:
    gather did not start for pod must-gather-qq6mb: timed out waiting for the condition

Falling back to `oc adm inspect clusteroperators.v1.config.openshift.io` to collect basic cluster information.
error running backup collection: errors occurred while gathering data:
    [skipping gathering namespaces/openshift-dns due to error: one or more errors occurred while gathering pod-specific data for namespace: openshift-dns

    [one or more errors occurred while gathering container data for pod dns-default-d8szj:

    [Get "https://10.0.129.179:10250/containerLogs/openshift-dns/dns-default-d8szj/dns?previous=true&timestamps=true": dial tcp 10.0.129.179:10250: connect: connection refused, Get "https://10.0.129.179:10250/containerLogs/openshift-dns/dns-default-d8szj/dns?timestamps=true": dial tcp 10.0.129.179:10250: connect: connection refused, Get "https://10.0.129.179:10250/containerLogs/openshift-dns/dns-default-d8szj/kube-rbac-proxy?previous=true&timestamps=true": dial tcp 10.0.129.179:10250: connect: connection refused, Get "https://10.0.129.179:10250/containerLogs/openshift-dns/dns-default-d8szj/kube-rbac-proxy?timestamps=true": dial tcp 10.0.129.179:10250: connect: connection refused], one or more errors occurred while gathering container data for pod dns-default-smqm2:

    [Get "https://10.0.165.230:10250/containerLogs/openshift-dns/dns-default-smqm2/dns?previous=true&timestamps=true": dial tcp 10.0.165.230:10250: connect: connection refused, Get "https://10.0.165.230:10250/containerLogs/openshift-dns/dns-default-smqm2/dns?timestamps=true": dial tcp 10.0.165.230:10250: connect: connection refused, Get "https://10.0.165.230:10250/containerLogs/openshift-dns/dns-default-smqm2/kube-rbac-proxy?previous=true&timestamps=true": dial tcp 10.0.165.230:10250: connect: connection refused, Get "https://10.0.165.230:10250/containerLogs/openshift-dns/dns-default-smqm2/kube-rbac-proxy?timestamps=true": dial tcp 10.0.165.230:10250: connect: connection refused], one or more errors occurred while gathering container data for pod node-resolver-nrclz:

    [Get "https://10.0.165.230:10250/containerLogs/openshift-dns/node-resolver-nrclz/dns-node-resolver?previous=true&timestamps=true": dial tcp 10.0.165.230:10250: connect: connection refused, Get "https://10.0.165.230:10250/containerLogs/openshift-dns/node-resolver-nrclz/dns-node-resolver?timestamps=true": dial tcp 10.0.165.230:10250: connect: connection refused], one or more errors occurred while gathering container data for pod node-resolver-zg6wj:

    [Get "https://10.0.129.179:10250/containerLogs/openshift-dns/node-resolver-zg6wj/dns-node-resolver?timestamps=true": dial tcp 10.0.129.179:10250: connect: connection refused, Get "https://10.0.129.179:10250/containerLogs/openshift-dns/node-resolver-zg6wj/dns-node-resolver?previous=true&timestamps=true": dial tcp 10.0.129.179:10250: connect: connection refused]], skipping gathering namespaces/openshift-etcd due to error: one or more errors occurred while gathering pod-specific data for namespace: openshift-etcd
```

**Comment by jhadvig** [comment]
QE Approver:
 /assign @yapei 
Docs Approver:
 /assign @opayne1  
PX Approver: 
/assign @RickJWagner

**Comment by yapei** [comment]
ack, @yanpzhan will test the PR

File: pkg/console/subresource/consoleserver/types.go [hunk 1]
```diff
@@ -54,9 +58,10 @@ type ServingInfo struct {
 
 // ClusterInfo holds information the about the cluster such as master public URL and console public URL.
 type ClusterInfo struct {
-	ConsoleBaseAddress string `yaml:"consoleBaseAddress,omitempty"`
-	ConsoleBasePath    string `yaml:"consoleBasePath,omitempty"`
-	MasterPublicURL    string `yaml:"masterPublicURL,omitempty"`
+	ConsoleBaseAddress  string                `yaml:"consoleBaseAddress,omitempty"`
+	ConsoleBasePath     string                `yaml:"consoleBasePath,omitempty"`
+	MasterPublicURL     string                `yaml:"masterPublicURL,omitempty"`
+	ControlPlaneToplogy configv1.TopologyMode `yaml:"controlPlaneTopology,omitempty"`
```
Comment (zherman0) [inline pkg/console/subresource/consoleserver/types.go:64]: @jhadvig - there is a spelling mistake here that does not effect things, but should probably be fixed:  `Topology`

**Pull request merged**